content/IA-update_connections/connection-state-and-recovery.textile

```diff
@@ -0,0 +1,292 @@
+---
+title: Connection state and recovery
+meta_description: "Establish and maintain a persistent connection to Ably using the Realtime Client Library SDK."
+section: realtime
+index: 20
+languages:
+  - javascript
+  - nodejs
+  - ruby
+  - java
+  - swift
+  - objc
+  - csharp
+  - flutter
+
+---
+
+The Ably Realtime library establishes and maintains a connection to the Ably service, using the most efficient transport available, typically "WebSockets":https://ably.com/topic/websockets. The Ably realtime protocol operates and multiplexes all channel traffic over that connection. Channels are separate from Connections and are used to logically separate messages into different topics, see the "channels documentation":/realtime/channels for further information.
+
+h2(#connection-state-explained). Connection state explained
+
+Although connection state is temporary, the Ably protocol provides continuity of message delivery between the client and the service, provided that a dropped connection is reestablished by the client within a limited interval (typically around 2 minutes). Beyond that, the connection becomes stale and the system will not attempt to recover the connection state. The lifecycle of a connection, and the strategy for reconnecting on failure, reflect the transient nature of the connection state.
```

Comment:
I would start by explaining what connection state is, and that a connection can be in multiple states such as...just to give a clearer description of what a connection state is.

Reply:
Fixed